Source/WebCore/platform/graphics/texmap/TextureMapperFPSCounter.cpp

```diff
@@ -64,12 +68,29 @@ void TextureMapperFPSCounter::updateFPSAndDisplay(TextureMapper& textureMapper,
         m_lastFPS = int(m_frameCount / delta.seconds());
         m_frameCount = 0;
         m_fpsTimestamp += delta;
+        if (measurementChangeCallback)
+            (*measurementChangeCallback)();
     }
 
     WTFSetCounter(FPS, m_lastFPS);
+    return true;
+}
 
-    if (m_isShowingFPS)
+void TextureMapperFPSCounter::updateFPSAndDisplay(TextureMapper& textureMapper, const FloatPoint& location, const TransformationMatrix& matrix)
+{
+    if (updateFPS())
         textureMapper.drawNumber(m_lastFPS, Color::black, location, matrix);
 }
 
+void TextureMapperFPSCounter::updateFPSAndLog(Page& page)
+{
+    updateFPS([this, &page]() {
+        if (m_isShowingFPS) {
+            page.forEachDocument([this](Document& document) {
```

Comment:
Do we want to log to all documents? wouldn't be enough to log to the main frame document?